lib/rails_rate_limiter.rb

```diff
@@ -73,4 +73,10 @@ def compute_requester_pattern(requester)
     # a wrong context that leads to not preventing action execution after render
     "custom_#{instance_exec(&requester)}"
   end
+
+  def compute_limit(limit)
+    return limit unless limit.respond_to?(:call)
+
+    instance_exec(&limit)
```

Comment:
without this functionality - the proc is always scoped to the context of the SlidingWindowLog object - which doesn't have access to anything for the current request. Allowing the limit proc to be within the context of the controller instance is far more powerful